When executing against a file encoded in Windows-31J (MS932), a java.nio.charset.UnmappableCharacterException occurs.

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using

- rewrite-core v8.51.0
- rewrite-java v8.51.0
- Maven/Gradle plugin v6.6.1
- java21

## How are you running OpenRewrite?
<!--
Are you using the Maven plugin, Gradle plugin, Moderne CLI, Moderne.io or something else?
Is your project a single module or a multi-module project?

Can you share your configuration so that we can rule out any configuration issues?

Is your project public? If so, can you share a link to it?
Code snippets can also be shared privately via [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).
-->
I am using the Maven plugin, and my project is a single module project.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>org.example</groupId>
    <artifactId>openrewrite</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>jar</packaging>

    <properties>
        <maven.compiler.source>21</maven.compiler.source>
        <maven.compiler.target>21</maven.compiler.target>
        <project.build.sourceEncoding>MS932</project.build.sourceEncoding>
        <project.reporting.outputEncoding>MS932</project.reporting.outputEncoding>
        <maven.compiler.parameters>true</maven.compiler.parameters>
    </properties>

    ... 
            <plugin>
                <groupId>org.openrewrite.maven</groupId>
                <artifactId>rewrite-maven-plugin</artifactId>
                <version>6.6.1</version>
                <configuration>
                    <activeRecipes>
                        <recipe>com.example.ChangeMethodAccessLevel</recipe>
                    </activeRecipes>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.example</groupId>
                        <artifactId>openrewrite</artifactId>
                        <version>1.0-SNAPSHOT</version>
                    </dependency>
                </dependencies>
            </plugin>
```

## What is the smallest, simplest way to reproduce the problem?
<!--
Sometimes the logs indicate a recipe stumbled over a particular pattern of code.
If you can share a code snippet that reproduces the issue, that will help us fix it faster.
We also accept [pull requests that merely replicate an issue](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes), as a step up to a full fix.

A code snippet can be something simple like this, or similar for other languages:
-->
- The file encoding is set to Windows-31J (MS932).

```java
package org.example;

public class TestClass {
    public void test() {
        // TODO: メソッド実行
        return;
    }
}
```

## What did you expect to see?
<!-- A code snippet, or a description of the behavior you expected helps us write a test to ensure the issue is fixed. -->
```java
package org.example;

public class TestClass {
    private void test() {
        // TODO: メソッド実行
        return;
    }
}
```

## What did you see instead?
<!-- A code snippet, or a description of the behavior you saw instead of the above expected result. -->
```java
package org.example;

public class TestClass {
    private void test() {
        // TODO: �
```

- If you're unable to reproduce the issue, please try using this project.
  - [openrewrite.zip](https://github.com/user-attachments/files/20008556/openrewrite.zip)

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:6.6.1:run (default-cli) on project openrewrite: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:6.6.1:run failed: Unable to rewrite source files: Input length = 1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:6.6.1:run (default-cli) on project openrewrite: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:6.6.1:run failed: Unable to rewrite source files
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
at java.lang.reflect.Method.invoke (Method.java:580)
at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
...
Caused by: java.nio.charset.UnmappableCharacterException: Input length = 1
at java.nio.charset.CoderResult.throwException (CoderResult.java:280)
at sun.nio.cs.StreamEncoder.implWrite (StreamEncoder.java:384)
at sun.nio.cs.StreamEncoder.implWrite (StreamEncoder.java:357)
at sun.nio.cs.StreamEncoder.lockedWrite (StreamEncoder.java:158)
at sun.nio.cs.StreamEncoder.write (StreamEncoder.java:139)
at java.io.OutputStreamWriter.write (OutputStreamWriter.java:219)
at java.io.BufferedWriter.implFlushBuffer (BufferedWriter.java:178)
at java.io.BufferedWriter.flushBuffer (BufferedWriter.java:163)
at java.io.BufferedWriter.implClose (BufferedWriter.java:397)
at java.io.BufferedWriter.close (BufferedWriter.java:380)
at org.openrewrite.maven.AbstractRewriteRunMojo.writeAfter (AbstractRewriteRunMojo.java:215)
at org.openrewrite.maven.AbstractRewriteRunMojo.execute (AbstractRewriteRunMojo.java:165)
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
at java.lang.reflect.Method.invoke (Method.java:580)
at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

## Context
- As a Japanese developer working on a legacy system migration, I am planning to use OpenRewrite.
- I am unable to change the file encoding of the target files, which are encoded in Windows-31J (MS932).
- The issue does not occur during JUnit test execution; the test completes successfully. However, it fails when using the Maven plugin.
```
mvn -X rewrite:run
```

--- a/src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java
+++ b/src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java
@@ -55,6 +55,7 @@ import org.openrewrite.maven.tree.ProfileActivation;
 import org.openrewrite.maven.tree.ResolvedGroupArtifactVersion;
 import org.openrewrite.maven.utilities.MavenWrapper;
 import org.openrewrite.style.NamedStyles;
+import org.openrewrite.tree.ParsingExecutionContextView;
 import org.openrewrite.xml.tree.Xml;
 
 import java.io.File;
@@ -182,7 +183,10 @@ public class MavenMojoProjectParser {
         JavaParser.Builder<? extends JavaParser, ?> javaParserBuilder = JavaParser.fromJavaVersion()
                 .styles(styles)
                 .logCompilationWarningsAndErrors(false);
-        getCharset(mavenProject).ifPresent(javaParserBuilder::charset);
+        getCharset(mavenProject).ifPresent(charset -> {
+            ParsingExecutionContextView.view(ctx).setCharset(charset);
+            javaParserBuilder.charset(charset);
+        });
 
         // todo, add styles from autoDetect
         KotlinParser.Builder kotlinParserBuilder = KotlinParser.builder();

